src/index.js

```diff
@@ -1,88 +1,115 @@
-                "<p class='detection-alert__text'>The page you're trying to view isn't optimised for the browser version you're using. It's possible that you will experience some problems with the layout and potentially some errors.</p>" +
-                "<div><button class='detection-alert__button' onclick=\"{" +
-            "window.sessionStorage && window.sessionStorage.setItem('outdated-browser', true);" +
-            "document.body.removeChild(document.getElementById('detection-alert'));" +
-            "document.body.removeChild(document.getElementById('detection-overlay'));}\">OK</button></div>";
-        t = t.en;
-
-        var div = document.createElement("div");
-        div.className = "detection-alert";
-        div.id = "detection-alert";
-
-        var div_overlay = document.createElement("div");
-        div_overlay.className = "detection-overlay";
-        div_overlay.id = "detection-overlay";
-
-        t = t.replace("{brow_name}", browserInfo.name);
-
-        div.innerHTML = '<div id="detection-div">' + t + '</div>';
-        document.body.insertBefore(div, document.body.firstChild);
-        document.body.insertBefore(div_overlay, document.body.firstChild);
+    var bdetector = {
+        'detection': dd
     };
 
-    var bdetector = {
-        'detector': dd
+    var detection = {
+        'detectBrowser': detectBrowser,
+        'transformVersion': transformVersion,
+        'compareDefaults': compareDefaults,
+        'outdated': outdated,
+        'show': show
     };
 
-    window.bdetector = bdetector;
+    window.detector = bdetector;
 
     if (typeof module == 'object' && module.exports) {
-        module.exports = bdetector;
+        module.exports = {bdetector: bdetector, detection: detection};
     }
 
-}(typeof window !== "undefined" ? window : this, document));
\ No newline at end of file
+}(typeof window !== "undefined" ? window : this, document));
+
+function detectBrowser(pattern, ua) {
+    for (var i = 0; i < pattern.length; i++) {
+        if (ua.match(new RegExp(pattern[i][0].replace("VV", "(\\d+\\.?\\d?)")), "i")) {
+            var browser = pattern[i][1];
+            break;
+        }
+    }
+    return {browser: browser, version: parseFloat(RegExp.$1)};
+};
+
+function transformVersion(browser, version) {
+    if (browser == "so") {
+        version = 4.0;
+        browser = "s";
+    }
+    if (browser == "io") {
+        browser = "i";
+        if (version > 6) version = 11;
+        else if (version > 5) version = 10;
+        else if (version > 4) version = 9;
+        else if (version > 3.1) version = 8;
+        else if (version > 3) version = 7;
+        else version = 9;
+    }
+    return {browser: browser, version: version};
+}
+
+function compareDefaults(vs, bvdefault) {
+    for (var brow in bvdefault) {
+        if (!vs[brow]) {
+            vs[brow] = bvdefault[brow];
+        }
+    }
+    return vs;
```

Comment:
vs is a object. When return vs is not necessary because object is changed by reference.